src/gluonts/itertools.py

```diff
@@ -174,6 +175,21 @@ def __iter__(self):
         yield from itertools.islice(self.iterable, self.length)
 
 
+class Map:
+    def __init__(self, fn, iterable: SizedIterable):
```

Comment:
This could be even more generic, accept also potentially a `Collection`, and expose `__getitem__`. But that complicates it, so maybe we do that if/when needed

Reply:
But `__getitem__` would only work if `iterable` is `Collection`. I don't think that's expressible in Python's type system.

Reply:
I was thinking of just not annotating it, but again this is not important as of now